apps/src/craft/designer/game/Entities/BaseEntity.js

```diff
@@ -496,4 +504,32 @@ export default class BaseEntity {
         return Math.abs(Math.pow(this.position[0] - entity.position[0], 2) + Math.pow(this.position[1] - entity.position[1], 2));
     }
 
+    blowUp(commandQueueItem, explosionPosition) {
+        let pushBackDirection = FacingDirection.Down;
+        if (explosionPosition[0] > this.position[0]) {
+            pushBackDirection = FacingDirection.Left;
+            this.facing = FacingDirection.Right;
+            this.updateAnimationDirection();
+        } else if (explosionPosition[0] < this.position[0]) {
+            pushBackDirection = FacingDirection.Right;
+            this.facing = FacingDirection.Left;
+            this.updateAnimationDirection();
+        } else if (explosionPosition[1] > this.position[1]) {
+            pushBackDirection = FacingDirection.Up;
+            this.facing = FacingDirection.Down;
+            this.updateAnimationDirection();
+        } else if (explosionPosition[1] < this.position[1]) {
+            pushBackDirection = FacingDirection.Down;
+            this.facing = FacingDirection.Up;
+            this.updateAnimationDirection();
+        }
+        this.pushBack(commandQueueItem, pushBackDirection, 150, function (entity) {
+            let callbackCommand = new CallbackCommand(entity.controller, () => { }, () => { entity.controller.destroyEntity(callbackCommand, entity.identifier); }, entity.identifier);
```

Comment:
nit (doesnt need to block merge): could be const instead of let, ideally wouldnt be one long line